pygmt/src/xyz2grd.py

```diff
@@ -0,0 +1,72 @@
+"""
+xyz2grd - Convert data table to a grid.
+"""
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    G="outgrid",
+    I="spacing",
+    R="region",
+    V="verbose",
+)
+@kwargs_to_strings(R="sequence")
+def xyz2grd(table, **kwargs):
+    """
+    Create a grid file from table data.
+
+    xyz2grd reads one or more z or xyz tables and creates a binary grid file.
+    xyz2grd will report if some of the nodes are not filled in with data. Such
+    unconstrained nodes are set to a value specified by the user [Default is
+    NaN]. Nodes with more than one value will be set to the mean value.
+
+    Full option list at :gmt-docs:`xyz2grd.html`
+
+
+    Parameters
+    ----------
+    table : str or {table-like}
+        Pass in either a file name to an ASCII data table, a 1D/2D
+        {table-classes}.
+
+    outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    {I}
+    {R}
+    {V}
+    Returns
```

Comment:
There's a whole list of common aliases we could add (`-b`, `-d`, `-e`, `-f`, `-h`, `-i`, `-r`, `-w`) since #1407, but for the sake of not stretching this PR any longer, maybe do it a separate time.

```suggestion
    {V}

    Returns
```

Reply:
I'm a fan of separating the PR, if only because I hate how the file closes and disappears when I switch branches right now :laughing: 

But seriously, I think it is better to have minimal PRs that get the function wrapped and then later PRs that add functionality. A bonus is that I think adding parameters is a good first issue for someone looking to make the jump from writing documentation/examples to editing the GMT wrapper.